### ISSUE TITLE
Add-BitbucketRepositoryBranchRestriction does not remove branch_type property

### ISSUE THREAD

**Issue by variableresistor** [beyondcomputing-org/Atlassian.Bitbucket#14]
This causes the error:
`branch_type is only valid when branch_match_kind is branching_model"`
When trying to add a merge check using the Add-BitbucketRepositoryBranchRestriction Cmdlet. To reproduce:
```
$ReferenceRepoSlug = "RefRepo"
$DestinationRepoSlug = "DiffRepo"

$BranchName = "development"
Get-BitbucketRepositoryBranchRestriction -RepoSlug $ReferenceRepoSlug | Where-Object -Property pattern -EQ $BranchName | Where-Object { $_.GetType().Name -eq "MergeCheck" } | ForEach-Object {
$Check = New-BitbucketRepositoryBranchRestrictionMergeCheck -Kind $_.kind -Pattern $_.pattern -Value $_.value
Add-BitbucketRepositoryBranchRestriction -RepoSlug $DestinationRepoSlug -Restriction $Check
}
```

The incorrect logic is https://github.com/beyondcomputing-org/Atlassian.Bitbucket/blob/b59de675ad4fed33fb104f9bd266041ad6c98d11/Atlassian.Bitbucket.Repository.BranchRestriction.psm1#L52
When excluding a property in this manner, you need to specify a wildcard for the -Property parameter. Will submit a PR in accordance with the contributing guidelines shortly.

**Comment by variableresistor** [comment]
#15

**Comment by eelkram** [comment]
Thanks @variableresistor.  I need to make some other updates for API deprecations but then I'll circle around and take a look at this as well.

**Comment by eelkram** [comment]
Thanks for the contribution @variableresistor .  The PR has been merged and version 0.29.1 published on the PS Gallery.
